docs/developers/bitesize/IoActor/IoActor.md

```diff
@@ -0,0 +1,36 @@
+# IoActor: basic concepts
+
+* **Word count:** 225
+
+## Actor Hierarchy
+
+IO subsystem consists of the following actors.
+
+![IoActor hierarchy](IoActor.png)
+
+All messages are sent to `IoActorProxy`, which then routes them either to `IoActor` or `IoPromiseProxyActor` based on if 
+message contains a Promise. `IoActorProxy` receives a message with Promise, forwards message to the `IoActor`, and once 
```

Comment:
(if you can fit the answer in your remaining words...) Why might a message contain a promise?

Reply:
I see you actually answer this in the `Message types and processing logic` section below. Does the "`IoActorProxy` receives a message with Promise..." duplicate what's already written below?

Reply:
I think that mentioning Promise right here helps to justify why we need a proxy actor at all, whereas in the `Message types and processing logic` I'm explaining what is that `clientContext` field which may be or may be not part of the `IoActors's` incoming message.